examples/cpp/authentication.cpp

```diff
@@ -0,0 +1,26 @@
+#include <catch2/catch_test_macros.hpp>
+#include <string>
+#include <future>
+
+#include <cpprealm/sdk.hpp>
+
+TEST_CASE("create and log in a user", "[realm]") {
+    // :snippet-start: register-user
+    const std::string APP_ID = "cpp-tester-uliix";
+    const std::string USER_EMAIL = "testUser@mongodb.com";
+    const std::string USER_PASSWORD = "password1234";
+
+    auto app = realm::App(APP_ID);
+
+    app.register_user(USER_EMAIL, USER_PASSWORD).get_future().get();
+    // :snippet-end:
+
+    // :snippet-start: log-user-in
+    auto user = app.login(realm::App::Credentials::username_password(USER_EMAIL, USER_PASSWORD))
+    .get_future().get();
```

Comment:
I'd expect an indentation for a continuation of the same statement:
```suggestion
        .get_future().get();
```